### ISSUE TITLE
Checkstyle config_loc not always set.

### ISSUE THREAD

**Issue by NielsDoucet** [gradle/gradle#11058]
The documentation for the checkstyle plugin references the config_loc variable in a way that suggests it will always be present. See https://docs.gradle.org/5.5.1/userguide/checkstyle_plugin.html#sec:checkstyle_built_in_variables
If the user does not specify a configDir to an existing location, or if the default location ($rootProject/config/checkstyle) does not exist, the plugin does not actually pass this property to the checkstyle ant task.
This results in the following exception to occur:
```
Caused by: org.xml.sax.SAXException: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property ${config_loc} has not been set
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property ${config_loc} has not been set
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader.startElement(ConfigurationLoader.java:657)
```

### Expected Behavior

The property should always be passed, so optional references can work without issue.

### Current Behavior

If the configDir path does not exist, the property is not passed in.

### Context

We adapted the code snippet from the documentation (referenced above) to
```
<!-- config_loc is a default property provided by the gradle checkstyle plugin -->
    <!-- This filter is optional, so we don't crash if the file is not provided.   -->
    <module name="SuppressionFilter">
        <property name="file" value="${config_loc}/suppressions.xml"/>
        <property name="optional" value="true"/>
    </module>
```
This should allow any consuming project of the checkstyle.xml file to optionally provide suppressions.

### Steps to Reproduce 

1. configure checkstyle to use config_loc in its configuration, but make sure the configDir doesn't exist
2. run `gradle checkstyleMain`

### Your Environment
gradle 5.5.1
checkstyle tooling version 8.24

**Comment by big-guy** [comment]
Thanks! This is fixed in 6.0 (which is due for a RC any day now).

Dupe of https://github.com/gradle/gradle/issues/9762

**Comment by NielsDoucet** [comment]
Great, thanks. Totally missed it when searching for an existing bug report. Sorry for the noise :)
